### PR TITLE
Change script context names for run time fields to type_field

### PR DESCRIPTION
**Pull request by jdconrad**
This change does two things:
1. renames all script contexts for runtime fields to type_field such as `long_field` or `geo_point_field`
2. disallows runtime fields scripts to be stored when a runtime fields context is specified as part of the stored script; this does not change a user's ability delete any stored script as context is only used during a put

Note that change 2 adds a boolean value to contexts to determine whether or not they are allowed to be stored. I added a new constructor for this and only changed it for runtime fields.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
Lgtm so long as you've manual tested an upgrade from a version which has one of these stored to a version with this code.

**Comment by jdconrad** [comment]
@nik9000 I have not done that, but I will! Thanks for taking a look.

File: server/src/main/java/org/elasticsearch/script/ScriptContext.java [hunk 1]
```diff
@@ -96,13 +99,21 @@ public ScriptContext(String name, Class<FactoryType> factoryClazz, int cacheSize
         this.cacheSizeDefault = cacheSizeDefault;
         this.cacheExpireDefault = cacheExpireDefault;
         this.maxCompilationRateDefault = maxCompilationRateDefault;
+        this.allowStoredScript = allowStoredScript;
+    }
+
+    /** Construct a context with the related instance and compiled classes with caller provided cache defaults and allow
+     * scripts of this context to be stored scripts */
+    public ScriptContext(String name, Class<FactoryType> factoryClazz, int cacheSizeDefault, TimeValue cacheExpireDefault,
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/script/ScriptContext.java:107]: I do not think we should add a new ctor. Instead, we should update the existing uses with the new parameter. I would separate this change out into its own PR.

**Review by rjernst — CHANGES_REQUESTED** [review]

**Comment by jdconrad** [comment]
Closing this and will split this into a PR for each change separately at @rjernst 's request.
